src/boot/efi/efi-string.c

```diff
@@ -953,6 +956,75 @@ void *memcpy(void * restrict dest, const void * restrict src, size_t n) {
         return dest;
 }
 
+void *memmove(void *dst, const void *src, size_t len) {
+        size_t i;
+
+        /*
+         * If the buffers don't overlap, it doesn't matter what direction
+         * we copy in. If they do, it does, so just assume they always do.
+         * We don't concern ourselves with the possibility that the region
+         * to copy might roll over across the top of memory, because it's
+         * not going to happen.
+         *
+         * If the destination is above the source, we have to copy
+         * back to front to avoid overwriting the data we want to
+         * copy.
+         *
+         *      dest:       dddddddd
+         *      src:    ssssssss   ^
+         *              |   ^  |___|
+         *              |___|
+         *
+         * If the destination is below the source, we have to copy
+         * front to back.
+         *
+         *      dest:   dddddddd
+         *      src:    ^   ssssssss
+         *              |___|  ^   |
+         *                     |___|
+         */
+
+        if ((uintptr_t)dst < (uintptr_t)src) {
+                /*
+                 * As author/maintainer of libc, take advantage of the
+                 * fact that we know memcpy copies forwards.
+                 */
+                return memcpy(dst, src, len);
+        }
+
+        /*
+         * Copy by words in the common case. Look in memcpy.c for more
+         * information.
+         */
+
+        if ((uintptr_t)dst % sizeof(long) == 0 &&
+            (uintptr_t)src % sizeof(long) == 0 &&
+            len % sizeof(long) == 0) {
```

Comment:
Don't bother with this. Just do the minimal fallback like memcpy and use call BS->CopyMem (yes, it's required to handle overlapping memory).

Reply:
got rid of memmove, resolving